Sources/GRPC/ClientCalls/ClientCall.swift

```diff
@@ -52,6 +52,15 @@ public protocol ClientCall {
   func cancel()
 }
 
+extension ClientCall {
+  public func waitForCompletion(errorCallback: (GRPCStatus) -> Void = { _ in }) {
```

Comment:
What's the advantage of this method over `try wait()`?

Reply:
Oops -- I didn't mean to commit this. I was playing around with ideas because waiting on the `GRPCStatus` can be a bit of a pain: even though we guarantee that it will never fail (i.e. `try! status.wait()` should not fail) it's something that users have to trust and is absolutely a code-smell. 

Also when the future is completed the user has to switch on the status code to check whether it's `.ok` or not. So there's actually quite a lot of boilerplate required to check the outcome of the call.

Not sold on any solution yet but I've been thinking that a never-failing future would be helpful (i.e. you would `recover` or similar into something with no `.failure` branch), as would futures with a particular error type (i.e. like `Result`).

Reply:
Sounds like we should revert this here and file an issue for a better future solution :-)